quill-cassandra-macros/src/main/scala/io/getquill/context/cassandra/encoding/CollectionsEncodingMacro.scala

```diff
@@ -0,0 +1,39 @@
+package io.getquill.context.cassandra.encoding
+
+import io.getquill.util.Messages._
+
+import scala.reflect.macros.blackbox.{ Context => MacroContext }
+
+trait CollectionsEncodingMacro {
```

Comment:
On the first glance it looked easier to implement because of the following:
For Cassandra encoders/decoders we need a some kind of mapper: 
- boxing/unboxing primitives (this needs to be done manually)
- identity function for supported types
- mapped encoding

For list and set we can implement something like with sql arrays - collection encoder for each base type. But for map?
One solution is to introduce some type class which will mark supported types and pass each into implicit context.  Also another encoders for AnyVals (oh, i completely forgot about anyvals) and primitives. And another one for mapping encoders. This looked a boilerplate for me for the first time. However I can try this wdyt?
Having such "CassandraType[T]" will allow as to create generic encoder using cassandra row.set[T]. Also we can create some mechanism to create custom CassandraType[T] in order to create cassandra codecs for it.
Wdyt?